revocation/crl/cache/bundle.go

```diff
@@ -0,0 +1,73 @@
+// Copyright The Notary Project Authors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cache
+
+import (
+	"crypto/x509"
+	"errors"
+	"time"
+)
+
+// CRLMetadata stores the URL of the CRL
+type CRLMetadata struct {
+	// URL stores the URL of the CRL
+	URL string `json:"url"`
+
+	// NextUpdate stores the next update time of the CRL
+	//
+	// This field extracts the `NextUpdate` field from the CRL extensions, which
+	// is an optional field in the CRL, so it may be empty.
+	NextUpdate time.Time `json:"nextUpdate"`
+}
+
+// Metadata stores the metadata infomation of the CRL
+//
+// TODO: consider adding DeltaCRL field in the future
+type Metadata struct {
+	// BaseCRL stores the URL of the base CRL
+	BaseCRL CRLMetadata `json:"baseCRL"`
+
+	// CachedAt stores the creation time of the CRL bundle. This is different
+	// from the `ThisUpdate` field in the CRL. The `ThisUpdate` field in the CRL
+	// is the time when the CRL was generated, while the `CachedAt` field is for
+	// caching purpose, indicating the start of cache effective period.
+	CachedAt time.Time `json:"cachedAt"`
```

Comment:
Why do we need `CachedAt` and `NextUpdate` time, they are already part of [x509#RevocationList](https://pkg.go.dev/crypto/x509#RevocationList)?

Reply:
`This Update` is also an optional extension, and many CRLs don't have this extension. We need a `CachedAt` field to save the download time, so we can know the update time for `DefaultMaxAge`, which is `CachedAt` + `DefaultMaxAge`